ql/ql/src/codeql_ql/ast/internal/Predicate.qll

```diff
@@ -77,8 +77,12 @@ private module Cached {
       p = t.getClassPredicate(mc.getMemberName(), mc.getNumberOfArguments())
     )
     or
-    // super calls
-    exists(Super sup, ClassType type, Type supertype |
+    // super calls - and `this.method()` calls in charpreds (for confusing reasons)
```

Comment:
I'm not really a big fan of "for confusing reasons". Nor do I think this is the right place to tell the whole story. Is this PR unblocking anything? If not, shouldn't we just wait and see the what Core team decides on this?